packages/react/src/ProgressBar/ProgressBar.tsx

```diff
@@ -60,37 +57,76 @@ const ProgressContainer = styled.span<StyledProgressContainerProps>`
   ${sx};
 `
 
+export type ProgressBarItems = React.HTMLAttributes<HTMLSpanElement> & {'aria-label'?: string} & ProgressProp & SxProp
+
+export const Item = forwardRef<HTMLSpanElement, ProgressBarItems>(
+  (
+    {progress, 'aria-label': ariaLabel, 'aria-valuenow': ariaValueNow, 'aria-valuetext': ariaValueText, ...rest},
+    forwardRef,
+  ) => {
+    const progressAsNumber = typeof progress === 'string' ? parseInt(progress, 10) : progress
+
+    const ariaAttributes = {
+      'aria-valuenow':
+        ariaValueNow || (progressAsNumber !== undefined && progressAsNumber >= 0 ? Math.round(progressAsNumber) : 0),
+      'aria-valuemin': 0,
+      'aria-valuemax': 100,
+      'aria-valuetext': ariaValueText,
+    }
+
+    return (
+      <ProgressItem
+        role="progressbar"
+        aria-label={ariaLabel}
+        ref={forwardRef}
+        progress={progress}
+        {...ariaAttributes}
+        {...rest}
+      />
+    )
+  },
+)
+
+Item.displayName = 'ProgressBar.Item'
+
 export type ProgressBarProps = React.HTMLAttributes<HTMLSpanElement> & {bg?: string} & StyledProgressContainerProps &
   ProgressProp
 
 export const ProgressBar = forwardRef<HTMLSpanElement, ProgressBarProps>(
   (
-    {animated, progress, bg = 'success.emphasis', barSize = 'default', children, ...rest}: ProgressBarProps,
+    {
+      animated,
+      progress,
+      bg = 'success.emphasis',
+      barSize = 'default',
+      children,
+      'aria-label': ariaLabel,
+      'aria-valuenow': ariaValueNow,
+      'aria-valuetext': ariaValueText,
+      ...rest
+    }: ProgressBarProps,
     forwardRef,
   ) => {
     if (children && progress) {
       throw new Error('You should pass `progress` or children, not both.')
     }
 
-    warning(
-      children &&
-        typeof (rest as React.AriaAttributes)['aria-valuenow'] === 'undefined' &&
-        typeof (rest as React.AriaAttributes)['aria-valuetext'] === 'undefined',
-      'Expected `aria-valuenow` or `aria-valuetext` to be provided to <ProgressBar>. Provide one of these values so screen reader users can determine the current progress. This warning will become an error in the next major release.',
-    )
-
-    const progressAsNumber = typeof progress === 'string' ? parseInt(progress, 10) : progress
-
-    const ariaAttributes = {
-      'aria-valuenow': progressAsNumber ? Math.round(progressAsNumber) : undefined,
-      'aria-valuemin': 0,
-      'aria-valuemax': 100,
-      'aria-valuetext': progressAsNumber ? `${Math.round(progressAsNumber)}%` : undefined,
-    }
+    const validChildren = React.Children.toArray(children).length
```

Comment:
Would https://react.dev/reference/react/Children#children-count work here as well for getting the length?

Reply:
I tried this but ran into an issue where non-components would be included in the count:

```tsx
<ProgressBar barSize="small" aria-label="Upload test.png">
  {false}
  {false}
  {false}
</ProgressBar>
  ```
  
This resulted in `React.Children.count(children)` to equal 3 😕 This isn't really a big deal, but I noticed this via a conditional in dotcom that was causing an error, and it was because the children were booleans instead of actual components 😅

Reply:
Oh wow, TIL https://react.dev/reference/react/Children#children-toarray:~:text=Empty%20nodes%20(null%2C%20undefined%2C%20and%20Booleans)%20will%20be%20omitted%20in%20the%20returned%20array 🤯 

Would it be possible to add a comment about this? I had no idea and wanted to suggest this in case it would help others out, maybe something like:

```suggestion
    // Get the number of non-empty nodes passed as children, this will exclude
    // booleans, null, and undefined
    const validChildren = React.Children.toArray(children).length
```

Reply:
Good idea, thank you for the suggestion and link 👀